lib/engine/game/g_2038/step/waterfall_auction.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/waterfall_auction'
+
+module Engine
+  module Game
+    module G2038
+      module Step
+        class WaterfallAuction < Engine::Step::WaterfallAuction
+          def buy_company(player, company, price)
+            super
+
+            return unless company.instance_of?(G2038::Company)
+
+            company.close!   # remove our wrapper which was added in super.buy_company
+            minor = @game.minors.find { |m| m.id == company.minor_id }
+            minor.owner = player
+            minor.float!
+            capital = (price - 100) / 2
+            minor.cash = 100 + capital
```

Comment:
You need to have the bank spend the money into the minor, not just set the treasury manually.